logic/common/log_utils.py

```diff
@@ -0,0 +1,111 @@
+''' Utility functions for printing and logging
+
+USAGE EXAMPLE:
+    import logic.common.utils_log as log
+    log.SetVerbosityLevel(1)
+    log.Info("Hello World!")
+
+# Verbosity level is usually first set in CLI file once during initialisation
+# Afterwards, you may use log.Info(s), log.Must(s), log.Extra(s) anywhere you need
+
+'''
+
+import os
+import logging
+
+#--------------------------------------------------#
+'''Logger Settings'''
+
+DEFAULT_FILENAME = 'logfile.log'
+DEFAULT_FORMAT = '%(message)s'   # Without extra indentation
+DEFAULT_DIVIDER = '--------------------------------------------------'
+
+FOLDER_PATH = 'reference/log/'
+
+
+
+#--------------------------------------------------#
+'''Configuration'''
+
+def SetVerbosityLevel(log_lv_num, file_name = DEFAULT_FILENAME, format = DEFAULT_FORMAT ):
+    '''Set basic configuration to logger'''
+    log_level = logging.WARNING # Default to Warning
+    if log_lv_num == 1: log_level = logging.INFO
+    elif log_lv_num == 2: log_level = logging.DEBUG
+
+    # Create a custom logger
+    logger = logging.getLogger()
+    logger.setLevel(log_level) # Set the minimum logging level
+
+    # Create a file handler to log messages to a file
+#    file_handler = logging.FileHandler(FOLDER_PATH+file_name)
+    file_handler = logging.FileHandler(FOLDER_PATH+file_name, 'w')
+    file_handler.setLevel(log_level)
+    file_handler.setFormatter(logging.Formatter(format))
+
+    # Create a stream handler for console output
+    console_handler = logging.StreamHandler()
+    console_handler.setLevel(log_level)
+    console_handler.setFormatter(logging.Formatter(format))
+
+    # Add the handlers to the logger
+    logger.addHandler(file_handler)
+    logger.addHandler(console_handler)
+    _WriteHeaderToLog(log_lv_num)
+
+
+
+def _WriteHeaderToLog(log_lv_num):
+    print_msg = ""
+    print_msg += "\n" + DEFAULT_DIVIDER + "\n"    # Divider before meta data
+    print_msg += "  "                             # Indentation
+    print_msg += os.getcwd() + "\n"               #  Current OS
+    print_msg += "  "                             # Indentation
+    print_msg += "Verbosity = " + str(log_lv_num) #  Verbosity Level
+    print_msg += "" # TODO DateTime
+    print_msg += ""
+    print_msg += "\n" + DEFAULT_DIVIDER           # Divider between meta data & tool content
+    print_msg += "\n"
+
+    # Header is logged regardless of verbosity
+    logging.warning(print_msg)
```

Comment:
Let's go for a reasonable amount of readability and compactness. Suggest:

`def _WriteHeaderToLog(log_lv_num):`
`    print_msg  = "\n" + DEFAULT_DIVIDER                 # Divider before meta data`
`    print_msg += "\n  " + os.getcwd()                   # Current OS`
`    print_msg += "\n  Verbosity = " + str(log_lv_num)   # Verbosity Level`
`    print_msg += "\n" + DEFAULT_DIVIDER + "\n"          # Divider between meta data & tool content`
`    #print_msg += "" # TODO DateTime`
`    logging.warning(print_msg)  # Header is logged regardless of verbosity`